### PR TITLE
Add docker micro benchmark into kubernetes contrib.

### PR DESCRIPTION
**Pull request by Random-Liu**
Add docker micro benchmark into kubernetes contrib.

The description and usage of the tool is in README.md.

Even though the tool is quite simple now, it could be a start. I'll add more benchmark when we need it in the future.

@kubernetes/sig-node

### CODE REVIEW

**Comment by yujuhong** [comment]
Still reviewing. Sorry for the delay.

**Comment by Random-Liu** [comment]
@yujuhong Thanks! Don't rush~ :)

**Comment by Random-Liu** [comment]
/cc @pwittrock @liangchenye 
This is a docker benchmark tool. Hope this could help you a bit. :)

**Comment by liangchenye** [comment]
:) Thanks

**Comment by yujuhong** [comment]
@Random-Liu, could you merge the first and the third commits? I'd like to go through the code once without the godeps changes.

**Comment by Random-Liu** [comment]
@yujuhong Sure. Will do soon.

**Comment by yujuhong** [comment]
@Random-Liu, thanks!

**Comment by Random-Liu** [comment]
@yujuhong Squashed 1st and 3rd commits.

File: docker-micro-benchmark/helpers/docker_helpers.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	wg.Wait()
+	allLatencies := []int{}
+	for _, latencies := range latenciesTable {
+		allLatencies = append(allLatencies, latencies...)
+	}
+	return allLatencies
+}
+
+// GetContainerIds returns all the container ids in the system
+func GetContainerIds(client *docker.Client) (containerIds []string) {
```
Comment (yujuhong) [inline docker-micro-benchmark/helpers/docker_helpers.go]: nit: s/Id/ID
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: docker-micro-benchmark/helpers/docker_helpers.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		name := newContainerName()
+		dockerOpts := docker.CreateContainerOptions{
+			Name: name,
+			Config: &docker.Config{
+				AttachStderr: false,
+				AttachStdin:  false,
+				AttachStdout: false,
+				Tty:          true,
+				Cmd:          []string{"/bin/bash"},
+				Image:        "ubuntu",
```
Comment (yujuhong) [inline docker-micro-benchmark/helpers/docker_helpers.go]: We'll want to make the image configurable in the future.

File: docker-micro-benchmark/benchmark_plans.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import (
+	"sync"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"k8s.io/contrib/docker-micro-benchmark/helpers"
+)
+
+var (
+	wg = &sync.WaitGroup{}
```
Comment (yujuhong) [inline docker-micro-benchmark/benchmark_plans.go]: unused?

File: docker-micro-benchmark/helpers/docker_helpers.go (hunk 1 above)
Comment (Random-Liu) [inline docker-micro-benchmark/helpers/docker_helpers.go]: @yujuhong Cool! Thanks a lot! Learned it~

File: docker-micro-benchmark/benchmark_plans.go (hunk 3 above)
Comment (Random-Liu) [inline docker-micro-benchmark/benchmark_plans.go]: @yujuhong Eh...Silly me...

File: docker-micro-benchmark/helpers/docker_helpers.go (hunk 2 above)
Comment (yujuhong) [inline docker-micro-benchmark/helpers/docker_helpers.go]: Actually, ubuntu is a pretty big image. Maybe we can use something more lightweight like alpine?

**Comment by yujuhong** [comment]
Let's get this in first and iterate later (if we need to). Thanks for putting this together!

**Comment by Random-Liu** [comment]
SGTM. I'll squash the commits. :) Thanks for reviewing~

**Pull request merged**